### PR TITLE
Improvement/CLDSRV-825 drop ceph support

### PR DESCRIPTION
**Pull request by benzekrimaha**
Issue: CLDSRV-825

### CODE REVIEW

**Comment by francoisferrand** [comment]
this PR seems to remove only the _tests_ : while the goal of the ticket was to remove supporting code (corner cases) as well:

> remove corner cases to support it (in production code!)

AFAIR, there were at least a few of these (which is consistent with the tests we had), can you double check?

File: lib/utilities/reportHandler.js [hunk 1]
```diff
@@ -72,7 +70,6 @@ function getCapabilities(cfg = config) {
         caps.locationTypeDigitalOcean &&= cfg.supportedLocationTypes.has('location-do-spaces-v1');
         caps.locationTypeSproxyd &&= cfg.supportedLocationTypes.has('location-scality-sproxyd-v1');
         caps.locationTypeNFS &&= cfg.supportedLocationTypes.has('location-nfs-mount-v1');
-        caps.locationTypeCephRadosGW &&= cfg.supportedLocationTypes.has('location-ceph-radosgw-s3-v1');
```
Comment (francoisferrand) [inline lib/utilities/reportHandler.js:75]: This is used to ensure consistency, Cloudserver does not have the "source of truth" : it comes from the ZenkoVersion CR (through ZKOP)
→ should not be kept
→ either we should keep this (to ensure consistency, even if ZenkoVersion indicates the version is supported), or enforce the new lack-of-support of cloudserver (i.e. remove 'location-ceph-radosgw-s3-v1' from the reported location types and unset locationTypeCephRadosGW)

File: tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAwsVersioning.js [hunk 2]
```diff
@@ -206,7 +205,7 @@ describeSkipIfNotMultiple('AWS backend delete object w. versioning: ' +
                     key, versionId: 'null', resultType: deleteVersion },
                     err => next(err, awsVerId)),
                 (awsVerId, next) => {
-                    const wanted = isCEPH ? 'NoSuchKey' : 'NoSuchVersion';
+                    const wanted = 'NoSuchVersion';
```
Comment (francoisferrand) [inline tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAwsVersioning.js]: should be inlined, no reason to keep the variable now

File: tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAzure.js [hunk 3]
```diff
@@ -30,7 +29,7 @@ const nonExistingId = process.env.AWS_ON_AIR ?
     'MhhyTHhmZ4cxSi4Y9SMe5P7UJAz7HLJ9' :
     '3939393939393939393936493939393939393939756e6437';
 
-describeSkipIfNotMultipleOrCeph('Multiple backend delete object from Azure',
+describe.skip('Multiple backend delete object from Azure',
```
Comment (francoisferrand) [inline tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAzure.js]: should not be skipped with multi-backend, it seems definition of `describeSkipIfNotMultipleOrCeph` was broken at some point → describeSkipIfNotMultiple

(if it does not work, create a followup ticket and reference it here)

Comment (francoisferrand) [reply tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAzure.js]: This applies to every place where `describeSkipIfNotMultipleOrCeph` was used, please fix them all to use `describeSkipIfNotMultiple` instead

File: tests/functional/aws-node-sdk/test/multipleBackend/objectPutCopyPart/objectPutCopyPartAzure.js [hunk 4]
```diff
@@ -126,7 +124,7 @@ function assertCopyPart(infos, cb) {
     ], cb);
 }
 
-describeSkipIfNotMultipleOrCeph('Put Copy Part to AZURE', function describeF() {
+describe.skip('Put Copy Part to AZURE', function describeF() {
```
Comment (francoisferrand) [inline tests/functional/aws-node-sdk/test/multipleBackend/objectPutCopyPart/objectPutCopyPartAzure.js]: same, should be be run with multiple data backend (even if it uses another definition of `describeSkipIfNotMultipleOrCeph` which you removed from line 23)

File: tests/functional/aws-node-sdk/test/multipleBackend/put/putGcp.js [hunk 5]
```diff
@@ -235,7 +235,7 @@ describeFn() {
     });
 });
 
-describeSkipIfNotMultipleOrCeph('MultipleBackend put object' +
+describe.skip('MultipleBackend put object' +
                                 'based on bucket location', () => {
```
Comment (francoisferrand) [inline tests/functional/aws-node-sdk/test/multipleBackend/put/putGcp.js]:
```suggestion
describe.skip('MultipleBackend put object based on bucket location', () => {
```

File: tests/functional/aws-node-sdk/test/object/websiteGetWithACL.js [hunk 6]
```diff
@@ -26,14 +26,12 @@ const aclEquivalent = {
 };
 
 const aclTests = [
-    // CEPH: test_website_private_bucket_list_private_index_blockederrordoc
```
Comment (francoisferrand) [inline tests/functional/aws-node-sdk/test/object/websiteGetWithACL.js:29]: this is weird ; I don't understand what these comments stand for, are they really related to Ceph?

**Review by francoisferrand — CHANGES_REQUESTED** [review]
- as already said, need to remove specific tweaks from the actual code (api handlers...)
- some tests are always skipped (not sure why) while they should run for multiple backend

File: tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAzure.js (hunk 3 above)
Comment (francoisferrand) [reply tests/functional/aws-node-sdk/test/multipleBackend/delete/deleteAzure.js]: note: this was introduced in https://github.com/scality/cloudserver/pull/4155/changes/c3385a4225bd28ced47511646f7f6ac202a70a4b, with not much explanation either in commit, PR or [ticket](https://scality.atlassian.net/browse/CLDSRV-93)

File: tests/functional/aws-node-sdk/test/object/websiteGetWithACL.js (hunk 6 above)
Comment (benzekrimaha) [reply tests/functional/aws-node-sdk/test/object/websiteGetWithACL.js:29]: Those comments weren’t about Ceph-specific behavior in Cloudserver IMO. They were legacy cross-references to Ceph’s S3 test case names (e.g. test_website_private_bucket_list_private_index_blockederrordoc), added when aligning our website ACL tests with theirs. 
The cases themselves are generic website endpoint + ACL tests (bucket/index/error doc visibility). The “CEPH:” prefix was only a test-name reference and was confusing, so I've removed them.

**Review by DarkIsDude — CHANGES_REQUESTED** [review]
Can you check the CI 🙏 ?

File: lib/utilities/reportHandler.js (hunk 1 above)
Comment (DarkIsDude) [reply lib/utilities/reportHandler.js:75]: I'm not sure to understand your point @francoisferrand

**Review by DarkIsDude — COMMENTED** [review]
I found some CEPH reference in the product doc that we should also clean. The CI is still unstable

**Review by DarkIsDude — APPROVED** [review]

File: lib/Config.js [hunk 7]
```diff
@@ -1880,7 +1874,9 @@ class Config extends EventEmitter {
 
         if (config.capabilities) {
             if (config.capabilities.locationTypes) {
-                this.supportedLocationTypes = new Set(config.capabilities.locationTypes);
+                this.supportedLocationTypes = new Set(
```
Comment (francoisferrand) [inline lib/Config.js]: why?
this should not be needed, unless there is an issue in tests.... so the tests should be updated instead: we should not adapt production code to tests, but tests to match how it happens in production

File: tests/multipleBackend/multipartUpload.js [hunk 8]
```diff
@@ -531,7 +523,7 @@ describe('Multipart Upload API with AWS Backend', function mpuTestSuite() {
                 })).then(() => {
                     assert.fail('Expected an error listing parts of aborted MPU');
                 }).catch(err => {
-                    const wantedError = isCEPH ? 'NoSuchKey' : 'NoSuchUpload';
+                    const wantedError = 'NoSuchUpload';
```
Comment (francoisferrand) [inline tests/multipleBackend/multipartUpload.js]: variable probably not needed here

File: tests/unit/utils/reportHandler.js [hunk 9]
```diff
@@ -38,9 +37,12 @@ describe('reportHandler.getCapabilities', () => {
                 secureChannelOptimizedPath: true,
                 s3cIngestLocation: true,
                 nfsIngestLocation: false,
-                cephIngestLocation: false,
                 awsIngestLocation: false,
             });
+            assert.strictEqual(
+                Object.prototype.hasOwnProperty.call(caps, 'locationTypeCephRadosGW'),
```
Comment (francoisferrand) [inline tests/unit/utils/reportHandler.js]: useless: redundant with `assert.deepStrictEqual` above, can never fail...

File: tests/unit/utils/reportHandler.js [hunk 10]
```diff
@@ -63,6 +65,31 @@ describe('reportHandler.getCapabilities', () => {
                 locationTypeS3Custom: false,
                 customCapability: 'test-value',
             });
+            assert.strictEqual(
+                Object.prototype.hasOwnProperty.call(caps, 'locationTypeCephRadosGW'),
```
Comment (francoisferrand) [inline tests/unit/utils/reportHandler.js]: useless: redundant with assert.deepStrictEqual above, can never fail...

**Review by francoisferrand — COMMENTED** [review]
a few small things, but more importantly it seems to me this cleanup is not complete: this is modifying only the _tests_, but we keep the code handling specific corner cases for cepth (which may be either/both in cloudserver and arsenal).

this code _MUST_ be cleaned as well, otherwise we just continue maintaining the same code, just with less coverage

**Comment by benzekrimaha** [comment]
@francoisferrand  Thanks for the careful review — I agree with your point that removing tests without removing related code would leave us with untested legacy behavior.
I re-audited this PR specifically with that objective:
I listed CEPH-conditioned test removals (isCEPH / itSkipCeph / describeSkipIfCeph / CEPH-only expected error variants).
I mapped those removed tests to the corresponding runtime paths.
I removed the CEPH-specific runtime handling that was still present.
What was actually CEPH-specific in CloudServer:
=> Capability-key cleanup in reportHandler getCapabilities() (locationTypeCephRadosGW / cephIngestLocation) was the remaining CEPH-specific logic.
This has now been removed, and the matching unit test coverage was removed accordingly.
**_What remains and why:_**
The remaining behaviors touched by removed CEPH-conditioned tests are generic S3/MPU API semantics, not CEPH branches. Examples:
Internal MPU metadata NoSuchKey mapped to public NoSuchUpload.
Multipart delete behavior controlled by legacyAwsBehavior.
Standard versioning/object-lock semantics (NoSuchKey / NoSuchVersion / MethodNotAllowed).
These paths are still covered by active non-CEPH tests.
**So this is not a tests-only cleanup anymore:**
CEPH-specific runtime handling in CloudServer has been removed.
Remaining logic is generic compatibility behavior with active coverage.
_If you want, I can also do the same explicit mapping pass on Arsenal in a follow-up to ensure both repos are aligned._

File: lib/utilities/reportHandler.js [hunk 11]
```diff
@@ -74,9 +74,6 @@ function getCapabilities(cfg = config) {
         caps.locationTypeLocal &&= cfg.supportedLocationTypes.has('location-file-v1');
     }
 
-    delete caps.locationTypeCephRadosGW;
-    delete caps.cephIngestLocation;
```
Comment (francoisferrand) [inline lib/utilities/reportHandler.js:78]: this was an earlier comment, we need to keep these!
cloudserver will not support ceph anymore, so we don't want to advertise that we still support it

**Comment by francoisferrand** [comment]
> Capability-key cleanup in reportHandler getCapabilities() (locationTypeCephRadosGW / cephIngestLocation) was the remaining CEPH-specific logic.

this is not "ceph-specific" logic, this is logic which was introduced earlier on purpose to correctly report that we DO NOT support ceph anymore. So it must be kept.

> I can also do the same explicit mapping pass on Arsenal in a follow-up to ensure both repos are aligned.

yes, as discussed on Monday we need to check in Arsenal as well : where the actual specifics of each (data) backend are implemented

**Review by francoisferrand — APPROVED** [review]
